src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -741,9 +741,18 @@ private IEnumerable<Library> CreateLibraries(IEnumerable<TargetLibrary>? librari
             {
                 return Enumerable.Empty<Library>();
             }
-            return libraries
-                .Select(property => CreateLibrary(property, runtime, libraryStubs))
-                .Where(library => library != null)!;
+
+            return CreateLibrariesNotNull(libraries, runtime, libraryStubs);
+        }
+
+        private IEnumerable<Library> CreateLibrariesNotNull(IEnumerable<TargetLibrary> libraries, bool runtime, Dictionary<string, LibraryStub>? libraryStubs)
+        {
+            foreach (TargetLibrary library in libraries)
+            {
+                Library? createdLibrary = CreateLibrary(library, runtime, libraryStubs);
+                if (createdLibrary is not null)
+                    yield return createdLibrary;
+            }
```

Comment:
Why not inline the foreach loop into L745 as the method is only used once?

Reply:
I would propose to make `CreateLibrariesNotNull` a local function so that no object is `new`ed if `libraries` is `null`, similar to the original logic.